back/app/serializers/web_api/v1/phase_serializer.rb

```diff
@@ -11,37 +11,36 @@ class WebApi::V1::PhaseSerializer < WebApi::V1::BaseSerializer
     :reacting_dislike_enabled, :reacting_dislike_method, :reacting_dislike_limited_max,
     :allow_anonymous_participation, :presentation_mode, :ideas_order, :input_term
 
-  with_options if: proc { |phase|
-    phase.voting?
-  } do
-    attribute :voting_method
-    attribute :voting_max_total
-    attribute :voting_min_total
-    attribute :voting_max_votes_per_idea
-    attribute :baskets_count
-    attribute :voting_term_singular_multiloc do |phase|
-      phase.voting_term_singular_multiloc_with_fallback
-    end
-    attribute :voting_term_plural_multiloc do |phase|
-      phase.voting_term_plural_multiloc_with_fallback
-    end
-  end
-
-  with_options if: proc { |phase|
-    phase.native_survey?
-  } do
-    attribute :native_survey_title_multiloc
-    attribute :native_survey_button_multiloc
+  %i[
+    voting_method voting_max_total voting_min_total
+    voting_max_votes_per_idea baskets_count
+    native_survey_title_multiloc native_survey_button_multiloc
+  ].each do |attribute_name|
+    attribute attribute_name, if: proc { |phase|
+      Factory.instance.participation_method_for(phase).supports_serializing?(attribute_name)
+    }
   end
 
   attribute :votes_count, if: proc { |phase, params|
-    phase.voting? \
```

Comment:
I wonder whether we couldn't make less of these attributes conditional. I understand they're not relevant for certain methods, but then the FE won't use them either.

Reply:
I was wondering as well. @IvaKop Do you see any potential issue in the frontend side if we always include the attributes that are currently conditional depending on the participation method? If not, then let's drop the conditionality?

The only potential risk I see is performance, as e.g. the `votes_count` is using the `TimelineService`.

Reply:
Feels like a logical step. I can't really answer that without checking everything in detail but I would say we could do it in a separate PR and run the e2e tests + maybe manually check a bit. I don't think it's critical for right now, though.